frontend/its-our-call/src/components/BookmarkStar.tsx

```diff
@@ -11,21 +15,81 @@ interface BookmarkStarProps {
 
 interface BookmarkStarState {
   isBookmarked: boolean;
+  isShowingPopover: boolean;
+  isLoadedSuggestions?: boolean;
+  suggestions: BookmarkSuggestionProps[];
+}
+
+interface BookmarkSuggestionProps {
+  startsBookmarked: boolean;
+  title: string;
+}
+
+interface BookmarkSuggestionState {
+  isSubBookmarked: boolean;
+}
+
+class BookmarkSuggestion extends React.Component<BookmarkSuggestionProps, BookmarkSuggestionState> {
+  constructor(props: BookmarkSuggestionProps) {
+    super(props);
+    this.state = {
+      isSubBookmarked: this.props.startsBookmarked === true,
+    };
+    this.toggleSubBookmark = this.toggleSubBookmark.bind(this);
+  }
+
+  toggleSubBookmark() {
+    this.setState({isSubBookmarked: !this.state.isSubBookmarked});
+    // TODO: prompt other ui callbacks
+  }
+
+  render() {
+    let starIcon = (<IoIosStarOutline />);
+    let bookmarkClass = 'unchecked';
+    if (this.state.isSubBookmarked) {
+      starIcon = (<IoIosStar />);
+      bookmarkClass = 'checked';
+    }
+    return (
+      <div className="suggestion">
+        <div className={'embedded-bookmark-star ' + bookmarkClass} onClick={this.toggleSubBookmark}>{starIcon}</div>
+        <div className="title">{this.props.title}</div>
+      </div>
+    );
+  }
 }
 
 class BookmarkStar extends React.Component<BookmarkStarProps, BookmarkStarState> {
   constructor(props: BookmarkStarProps) {
     super(props);
     this.state = {
       isBookmarked: this.props.isBookmarked === true,
+      isShowingPopover: false,
+      isLoadedSuggestions: true,
+      suggestions: [
+        {startsBookmarked: false, title: 'Example one (Senator, VT)'},
+        {startsBookmarked: false, title: 'Example two (Representative 9th District, VT)'}
+      ],
     };
     this.toggleBookmark = this.toggleBookmark.bind(this);
+    this.onClickOutside = this.onClickOutside.bind(this);
   }
   toggleBookmark() {
     this.setState({isBookmarked: !this.state.isBookmarked});
+    this.setState({isShowingPopover: !this.state.isShowingPopover});
     // TODO: prompt other ui callbacks
   }
 
+  onClickOutside(e: MouseEvent) {
+      if (e.target instanceof SVGSVGElement) {
+        return;
+        // the svg icon triggers onClickOutside, even on first click
+        // early exit to prevent the popover from immediately hiding
+        // TODO: fix this to only exit on the embedded svg of *this* component
```

Comment:
It seems like clicking in some parts of the star causes `isBookmarked` to change, but doesn't trigger the popover to appear. I think these are pixels that are part of the div containing `starIcon`, but not the icon itself, meaning the `toggleBookmark`  function gets called, but e.target `instanceof SVGSVGElement` evaluates to false. Can we get rid of this area?

Reply:
I think you found a literal edge case (clicking the path of the svg ignores the onclick for some reason.) 
![edge_case](https://user-images.githubusercontent.com/954150/38772498-4259e27e-4006-11e8-9bbb-e1fbd62b98c6.gif)
I have no clear reason why that happens, but I'm going to blame the react-tiny-popover module not getting along with the react-icons module.

I'll make this TODO solution NOWDO solution, since this will also handle that case as well.